src/client/QXmppCall.cpp

```diff
@@ -0,0 +1,739 @@
+///
+
+bool QXmppCallPrivate::sendAck(const QXmppJingleIq &iq)
+{
+    QXmppIq ack;
+    ack.setId(iq.id());
+    ack.setTo(iq.from());
+    ack.setType(QXmppIq::Result);
+    return manager->client()->sendPacket(ack);
+}
+
+bool QXmppCallPrivate::sendInvite()
+{
+    // create audio stream
+    QXmppCallStream *stream = findStreamByMedia(AUDIO_MEDIA);
+    Q_ASSERT(stream);
+
+    QXmppJingleIq iq;
+    iq.setTo(jid);
+    iq.setType(QXmppIq::Set);
+    iq.setAction(QXmppJingleIq::SessionInitiate);
+    iq.setInitiator(ownJid);
+    iq.setSid(sid);
+    iq.addContent(localContent(stream));
+    return sendRequest(iq);
+}
+
+/// Sends a Jingle IQ and adds it to outstanding requests.
+///
+
+bool QXmppCallPrivate::sendRequest(const QXmppJingleIq &iq)
+{
+    requests << iq;
+    return manager->client()->sendPacket(iq);
+}
+
+void QXmppCallPrivate::setState(QXmppCall::State newState)
+{
+    if (state != newState) {
+        state = newState;
+        Q_EMIT q->stateChanged(state);
+
+        if (state == QXmppCall::ActiveState)
+            Q_EMIT q->connected();
+        else if (state == QXmppCall::FinishedState)
+            Q_EMIT q->finished();
+    }
+}
+
+/// Request graceful call termination
+
+void QXmppCallPrivate::terminate(QXmppJingleIq::Reason::Type reasonType)
+{
+    if (state == QXmppCall::DisconnectingState ||
+        state == QXmppCall::FinishedState)
+        return;
+
+    // hangup call
+    QXmppJingleIq iq;
+    iq.setTo(jid);
+    iq.setType(QXmppIq::Set);
+    iq.setAction(QXmppJingleIq::SessionTerminate);
+    iq.setSid(sid);
+    iq.reason().setType(reasonType);
+    sendRequest(iq);
+    setState(QXmppCall::DisconnectingState);
+
+    // schedule forceful termination in 5s
+    QTimer::singleShot(5000, q, SLOT(terminated()));
+}
+
+QXmppCall::QXmppCall(const QString &jid, QXmppCall::Direction direction, QXmppCallManager *parent)
+    : QXmppLoggable(parent)
+{
+    d = new QXmppCallPrivate(this);
+    d->direction = direction;
+    d->jid = jid;
+    d->ownJid = parent->client()->configuration().jid();
+    d->manager = parent;
+}
+
+QXmppCall::~QXmppCall()
+{
+    delete d;
+}
+
+/// Call this method if you wish to accept an incoming call.
+///
+
+void QXmppCall::accept()
+{
+    if (d->direction == IncomingDirection && d->state == ConnectingState) {
+        Q_ASSERT(d->streams.size() == 1);
+        QXmppCallStream *stream = d->streams.first();
+
+        // accept incoming call
+        QXmppJingleIq iq;
+        iq.setTo(d->jid);
+        iq.setType(QXmppIq::Set);
+        iq.setAction(QXmppJingleIq::SessionAccept);
+        iq.setResponder(d->ownJid);
+        iq.setSid(d->sid);
+        iq.addContent(d->localContent(stream));
+        d->sendRequest(iq);
+
+        // notify user
+        d->manager->callStarted(this);
+
+        // check for call establishment
+        d->setState(QXmppCall::ActiveState);
+    }
+}
+
+/// Returns the GStreamer pipeline.
+///
+/// \since QXmpp 1.2
```

Comment:
```suggestion
/// \since QXmpp 1.3
```

The \since-tags in this file still need to be updated.